system/middleware.go

```diff
@@ -98,6 +150,11 @@ func strInSlice(strs []string, str string) int {
 
 func (application *Application) ApplyCsrfProtection(c *web.C, h http.Handler) http.Handler {
 	fn := func(w http.ResponseWriter, r *http.Request) {
+		// disable CSRF for API requests
+		if c.Env["IsAPI"].(bool) {
```

Comment:
Like other middleware, this one has to come after ApplyIsAPI, so I think it should be noted and the potential panic handled.